src/Common/Database/QueryUtils.php

```diff
@@ -14,8 +14,135 @@
 
 namespace OpenEMR\Common\Database;
 
+use Webmozart\Assert\InvalidArgumentException;
+use Webmozart\Assert\Assert;
+
 class QueryUtils
 {
+    /**
+     * Usage:
+     *   $adminsPasswordHash = QueryUtils::getSingleScalarResultBy('users', 'password', ['username' => 'admin']);
+     *
+     * @throws InvalidArgumentException
+     * @throws SqlQueryException
+     */
+    public static function getSingleScalarResultBy(string $table, string $column, array $condition = []): null|string|int
+    {
+        Assert::notEmpty($condition);
+
+        return self::fetchSingleValue(
+            sprintf(
+                'SELECT %s FROM `%s` WHERE %s',
+                self::escapeColumnName($column, [$table]),
+                self::escapeTableName($table),
+                implode(' AND ', array_map(
+                    static fn (string $field, int|string|null $value): string => sprintf('`%s` = ?', self::escapeColumnName($field, [$table])),
```

Comment:
self::escapeColumnName can be "expensive" resource wise. If can figure out how to only call it once, then that would be idea. 
I'd rec escaping them at beginning of function (note they basically die/error out if do not pass; so it's rally a check rather than an escape)
And would do the table check first (since the column escape first checks the table, so if table fails more efficient to have it fail on doing table check first)

Reply:
something like at top:
```self::escapeColumnName($column, [$table]);
$escTableName = self::escapeTableName($table);
$escColumnName = self::escapeColumnName($column, [$table]);
```
(and then use the $esc* variables in the calls below that)

Reply:
hi @igormukhingmailcom , It's not typos we are worried about. It is sql-injection. Since there is not binding for tables/columns and they are not surrounded by quotes, the best mechanism to prevent sql injection has been whitelisting.